### PR TITLE
Release agg context in tests part 2

### PR DESCRIPTION
**Pull request by not-napoleon**
Continues the work from https://github.com/elastic/elasticsearch/pull/90540.  Turns out we weren't using `MockBigArrays` in a whole bunch of places, because the mock didn't correctly wrap over `withCircuitBreakerService()`.  You can see the corrected version of that method commented out here.  It's commented out so that hundreds of tests which are too sloppy with releasing memory don't start failing, but (as noted) I've included it to reduce the chance of future merge conflicts.

Besides that, this PR contains several small fixes to aggregations tests to do the right thing with memory.  These fixes are harmless to merge independently, and doing so lets us break this work into smaller chunks.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java [hunk 1]
```diff
@@ -28,49 +24,48 @@
...
 
-    private boolean needsScores(AggregationBuilder builder) throws IOException {
-        try (
-            Directory directory = newDirectory();
-            RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
-            DirectoryReader reader = indexWriter.getReader()
-        ) {
-            return createAggregator(builder, new IndexSearcher(reader), new KeywordFieldType("f")).scoreMode().needsScores();
-        }
+    private void needsScores(AggregationBuilder builder, boolean expected) throws IOException {
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java:62]: Maybe `assertNeedsScores` so its more obvious this asserts something.

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregatorTests.java [hunk 2]
```diff
@@ -13,20 +13,43 @@
...
 
 public class BucketsAggregatorTests extends AggregatorTestCase {
 
+    private List<AggregationContext> toRelease = new ArrayList<>();
+
+    @Override
+    protected AggregationContext createAggregationContext(IndexSearcher indexSearcher, Query query, MappedFieldType... fieldTypes)
+        throws IOException {
+        AggregationContext context = super.createAggregationContext(indexSearcher, query, fieldTypes);
+        // Generally, we should avoid doing this, but
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregatorTests.java:42]: "but this test doesn't test things to do with reduction so it should be safe here"

**Review by nik9000 — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregatorTests.java (hunk 2 above)
Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregatorTests.java:42]: I just sort of trailed off mid sentence there.  Not sure what distracted me.

Comment (nik9000) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregatorTests.java]: happens

**Pull request merged**